doc/reference/reference_lua/box_null.rst

```diff
@@ -0,0 +1,83 @@
+.. _box-once:
+
+-------------------------------------------------------------------------------
+                             Constant `box.NULL`
+-------------------------------------------------------------------------------
+
+There are some major problems with using a Lua **nil** value in tuples.
+For example: you can't correctly assess length of sparse array (an array with
+nullable values set to null) in case of using Lua **nil** value.
+
+**Example:**
+
+.. code-block:: tarantoolsession
+
+    tarantool> t = {0, nil, 1, 2, nil}
+    ---
+    ...
+
+    tarantool> t
+    ---
+    - - 0
+      - null
+      - 1
+      - 2
+    ...
+
+    tarantool> #t
+    ---
+    - 4
```

Comment:
`#t` is undefined behavior. I hope you've read my habr post https://habr.com/ru/company/mailru/blog/493642/

Reply:
You bet i did. We are using #t to show sparse arrays problems - including undefined behaviour. Added Note mentioning it is an undefined behaviour.

Reply:
I think that this example is a bit  clearer
```lua
tarantool> #{1, nil, 2}
---
- 1
...

tarantool> #{1, box.NULL, 2}
---
- 3
...
```

Reply:
it is shorter, yes. But without a variable this is harder to catch with a glance IMHO.  Also it is useful to show output of both variants of t.